src/OpenTelemetry/Trace/Internal/BroadcastActivityProcessor.cs

```diff
@@ -96,21 +96,9 @@ public override Task ForceFlushAsync(CancellationToken cancellationToken)
             return Task.WhenAll(tasks);
         }
 
-        public void Dispose()
+        protected override void Dispose(bool disposing)
         {
-            this.Dispose(true);
-        }
-
-        protected virtual void Dispose(bool disposing)
-        {
-            try
-            {
-                this.ShutdownAsync(CancellationToken.None).GetAwaiter().GetResult();
-            }
-            catch (Exception ex)
-            {
-                OpenTelemetrySdkEventSource.Log.SpanProcessorException(nameof(this.Dispose), ex);
-            }
+            base.Dispose(disposing);
```

Comment:
Read [this](https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose#implement-the-dispose-pattern-for-a-derived-class) - the base class dispose should happen at the very end.

Reply:
i see. thank you for the guideline! will update!